addons/mail/static/src/core/web/messaging_menu_patch.js

```diff
@@ -171,7 +171,7 @@ patch(MessagingMenu.prototype, {
         if (this.canPromptToInstall) {
             value++;
         }
-        if (this.shouldAskPushPermission) {
+        if (this.shouldAskPushPermission && !this.store.isNotificationPermissionDismissed) {
```

Comment:
`!this.store.isNotificationPermissionDismissed` could be moved in `shouldAskPushPermission`.
That way this is also taken into account in `hasPreviews`.